extensions/wikia/BannerNotifications/BannerNotificationsController.class.php

```diff
@@ -0,0 +1,291 @@
+<?php
+/**
+ * Handles adding and rendering of confirmations (green stripe below global nav) and notifications (blue bubble boxes just above the fold)
+ *
+ * @author Maciej Brencz
+ */
+
+class BannerNotificationsController extends WikiaController {
+
+	const SESSION_KEY = 'oasisConfirmation';
+
+	// confirmation types
+	const CONFIRMATION_CONFIRM = 'confirm'; // Green
+	const CONFIRMATION_NOTIFY = 'notify'; // Blue
+	const CONFIRMATION_ERROR = 'error'; // Red
+	const CONFIRMATION_WARN = 'warn'; // Yellow
+
+	// HTML of user messages notification (rendered by MW core -- skin variable)
+
+	public function init() {
+		$this->confirmation = null;
+		$this->confirmationClass = null;
+	}
+
+	/**
+	 * Add confirmation message to the user session (so it persists between redirects)
+	 */
+	public static function addConfirmation( $message, $type = self::CONFIRMATION_CONFIRM ) {
+		wfProfileIn( __METHOD__ );
+
+		if ( !empty( $message ) ) {
+			$_SESSION[self::SESSION_KEY] = array(
+				'message' => $message,
+				'type' => $type,
+			);
+
+			wfDebug( __METHOD__ . " - {$message}\n" );
+		}
+
+		wfProfileOut( __METHOD__ );
```

Comment:
Let's rip these `wfProfileOut` `wfProfileIn` calls out since we don't need them anymore.